pkg/database/vercode.go

```diff
@@ -62,6 +62,10 @@ type VerificationCode struct {
 	LongExpiresAt time.Time
 	IssuingUserID uint
 	IssuingAppID  uint
+
+	// When we claim a VerificationCode through the API, we need to trigger
+	// the per-realm stats update.
+	AreClaiming bool
```

Comment:
not sure I understand why this is needed in addition to claimed?

Reply:
Seth had wanted to use Hooks for updating the stats counts, removing the straight DB update someplace else.

If we do that our ORM doesn't know about dirty bits. So, if the VerificationCode has Claimed set, we'll update the stats whenever the hook is called. Now, we could query the DB, and get the old value -- but it's a question of timing. If we do it BeforeSave, we don't know that we're actually going to do the write. If we do it AfterSave, we don't know if Claimed was set BEFORE the update happened. It's a mess.  (This discussion ignores the fact that testing exercises the API in a manner the app never will.)

So, there's two ways out of it -- having this stupid variable, and continuing to use Hooks; or have a Claim function that does this. Hooks are nice because they're auto-updated and you never have to remember to call Claim. The function is nice because it doesn't have this crap floating around.

I am shooting the middle on all approaches here, and as you and Seth have the larger picture, I defer to your judgements (right now!).

Reply:
to properly use this as a dirty bit, don't we need to set AreClaiming = false when updating the stats?

otherwise this problem persists

or only set it to true when the stats are updated (actually I like that better, because it enables backfill on stats later if we need to do that)

Reply:
AreClaiming will default to false (go runtime) for all values read from the DB.

Backfill is a nice side effect I hadn't considered.

Reply:
except, we're saving it to the database

Reply:
I thought only values listed as "gorm" are saved!!!!

Reply:
argh

Reply:
Hmmmmmm. After seeing that we need to track an additional field, I don't think this is the best approach. I think adding a `verificationcode.Claim()` or `database.ClaimVerificationCode` would be cleaner. Sorry for the turmoil.

Reply:
To not have it be part of the database model, you have to explicitly tag as ignore

https://gorm.io/docs/models.html#Fields-Tags